### ISSUE TITLE
[Bug]: Error: hcloud/inlineAttachServerToNetwork: attach server to network: provided IP is not available (ip_not_available)

### ISSUE THREAD

**Issue by sasoiliev** [hetznercloud/terraform-provider-hcloud#586]
### What happened?

The scenario goes like this:

1. Have a Terraform module with an `hcloud_network` and an `hcloud_server` that is part of the network by means of the inline `hcloud_server`'s `network` block (see the example code).
2. `terraform apply` the module,
3. Update something that forces replacement of the server (e.g., the image, the user data).
4. `terraform apply` the module again.

Observe the following error:

```
╷
│ Error: hcloud/inlineAttachServerToNetwork: attach server to network: provided IP is not available (ip_not_available)
│
│   with hcloud_server.test,
│   on main.tf line 26, in resource "hcloud_server" "test":
│   26: resource "hcloud_server" "test" {
│
╵
```

It seems that the inline syntax for specifying networks does not wait sufficiently for the network connection in the Hetzner backend to be released and afterwards it fails when it tries to re-create it, but that's just my guess.

I am able to reproduce this consistently.

Notably, I am not experiencing this issue when an explicit `hcloud_server_network` is used.

### What did you expect to happen?

I expected the operation to succeed both times.

### Please provide a minimal working example

```terraform
terraform {
  required_providers {
    hcloud = {
      source = "hetznercloud/hcloud"
    }
  }
}

resource "hcloud_network" "test" {
  name     = "test"
  ip_range = "172.30.1.0/24"
}

resource "hcloud_network_subnet" "test" {
  ip_range     = "172.30.1.0/24"
  network_id   = hcloud_network.test.id
  network_zone = "eu-central"
  type         = "cloud"
}

resource "hcloud_server" "test" {
  name        = "test"
  server_type = "cpx11"
  image       = "debian-10"
  location    = "fsn1"
  public_net {
    ipv4_enabled = false
    ipv6_enabled = false
  }
  network {
    network_id = hcloud_network.test.id
    ip         = "172.30.1.2"
  }
}
```

**Comment by apricote** [comment]
I think this is caused by the same underlying bug as #558. We do not correctly wait for the server delete action to complete, which causes these race conditions.

We are currently working on a fix for #558 and I will make sure to also test this usecase.

**Comment by apricote** [comment]
This was fixed in #590 and was released as `1.36.0`.

Feel free to reopen the issue or create a new one if you still have problems.
